CPU Kernel: Use per-operator headers

peterbell10 opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #71217
* #71216
* #71215
* #71214
* #71213
* #71212
* __->__ #71137

ATen has a header dependency problem. Whenever an operator is added or modified, it changes `ATen/Functions.h` and `ATen/NativeFunctions.h` which in turn requires essentially every single file to be rebuilt. Per-operator headers allow files to only include the specific operators they use and so minimizes unnecessary rebuilds during incremental builds and improves cache hits in CI builds.

See this note for more details:
https://github.com/pytorch/pytorch/blob/3a03af2f505494582f1359254431f0c47fd88705/aten/src/ATen/templates/Functions.h#L20

Differential Revision: [D33949900](https://our.internmc.facebook.com/intern/diff/D33949900)

peterbell10 committed:
CPU Kernel: Use per-operator headers

[ghstack-poisoned]

peterbell10 committed:
Update on "CPU Kernel: Use per-operator headers"




[ghstack-poisoned]

peterbell10 committed:
Update on "CPU Kernel: Use per-operator headers"




[ghstack-poisoned]

peterbell10 committed:
Update on "CPU Kernel: Use per-operator headers"




[ghstack-poisoned]

peterbell10 committed:
Update on "CPU Kernel: Use per-operator headers"




[ghstack-poisoned]

peterbell10 committed:
Update on "CPU Kernel: Use per-operator headers"




[ghstack-poisoned]

peterbell10 committed:
Update on "CPU Kernel: Use per-operator headers"




[ghstack-poisoned]

peterbell10 committed:
Rebase and fix merge conflicts on "CPU Kernel: Use per-operator headers"




[ghstack-poisoned]

peterbell10 committed:
Rebase and fix merge conflicts on "CPU Kernel: Use per-operator headers"


ATen has a header dependency problem. Whenever an operator is added or modified, it changes `ATen/Functions.h` and `ATen/NativeFunctions.h` which in turn requires essentially every single file to be rebuilt. Per-operator headers allow files to only include the specific operators they use and so minimizes unnecessary rebuilds during incremental builds and improves cache hits in CI builds.

See this note for more details:
https://github.com/pytorch/pytorch/blob/3a03af2f505494582f1359254431f0c47fd88705/aten/src/ATen/templates/Functions.h#L20

[ghstack-poisoned]

malfet commented:
@malfet has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D33949900).

peterbell10 committed:
Rebase on master on "CPU Kernel: Use per-operator headers"


ATen has a header dependency problem. Whenever an operator is added or modified, it changes `ATen/Functions.h` and `ATen/NativeFunctions.h` which in turn requires essentially every single file to be rebuilt. Per-operator headers allow files to only include the specific operators they use and so minimizes unnecessary rebuilds during incremental builds and improves cache hits in CI builds.

See this note for more details:
https://github.com/pytorch/pytorch/blob/3a03af2f505494582f1359254431f0c47fd88705/aten/src/ATen/templates/Functions.h#L20

Differential Revision: [D33949900](https://our.internmc.facebook.com/intern/diff/D33949900)

[ghstack-poisoned]

peterbell10 committed:
Rebase and fix merge conflicts on "CPU Kernel: Use per-operator headers"


ATen has a header dependency problem. Whenever an operator is added or modified, it changes `ATen/Functions.h` and `ATen/NativeFunctions.h` which in turn requires essentially every single file to be rebuilt. Per-operator headers allow files to only include the specific operators they use and so minimizes unnecessary rebuilds during incremental builds and improves cache hits in CI builds.

See this note for more details:
https://github.com/pytorch/pytorch/blob/3a03af2f505494582f1359254431f0c47fd88705/aten/src/ATen/templates/Functions.h#L20

Differential Revision: [D33949900](https://our.internmc.facebook.com/intern/diff/D33949900)

[ghstack-poisoned]

peterbell10 committed:
Rebase and fix merge conflicts on "CPU Kernel: Use per-operator headers"


ATen has a header dependency problem. Whenever an operator is added or modified, it changes `ATen/Functions.h` and `ATen/NativeFunctions.h` which in turn requires essentially every single file to be rebuilt. Per-operator headers allow files to only include the specific operators they use and so minimizes unnecessary rebuilds during incremental builds and improves cache hits in CI builds.

See this note for more details:
https://github.com/pytorch/pytorch/blob/3a03af2f505494582f1359254431f0c47fd88705/aten/src/ATen/templates/Functions.h#L20

Differential Revision: [D33949900](https://our.internmc.facebook.com/intern/diff/D33949900)

[ghstack-poisoned]

peterbell10 committed:
Rebase and fix merge conflicts on "CPU Kernel: Use per-operator headers"


ATen has a header dependency problem. Whenever an operator is added or modified, it changes `ATen/Functions.h` and `ATen/NativeFunctions.h` which in turn requires essentially every single file to be rebuilt. Per-operator headers allow files to only include the specific operators they use and so minimizes unnecessary rebuilds during incremental builds and improves cache hits in CI builds.

See this note for more details:
https://github.com/pytorch/pytorch/blob/3a03af2f505494582f1359254431f0c47fd88705/aten/src/ATen/templates/Functions.h#L20

Differential Revision: [D33949900](https://our.internmc.facebook.com/intern/diff/D33949900)

[ghstack-poisoned]

peterbell10 commented:
@malfet ping

peterbell10 committed:
Rebase and fix merge conflicts on "CPU Kernel: Use per-operator headers"


ATen has a header dependency problem. Whenever an operator is added or modified, it changes `ATen/Functions.h` and `ATen/NativeFunctions.h` which in turn requires essentially every single file to be rebuilt. Per-operator headers allow files to only include the specific operators they use and so minimizes unnecessary rebuilds during incremental builds and improves cache hits in CI builds.

See this note for more details:
https://github.com/pytorch/pytorch/blob/3a03af2f505494582f1359254431f0c47fd88705/aten/src/ATen/templates/Functions.h#L20

Differential Revision: [D33949900](https://our.internmc.facebook.com/intern/diff/D33949900)

[ghstack-poisoned]

peterbell10 committed:
Rebase and fix merge conflicts on "CPU Kernel: Use per-operator headers"


ATen has a header dependency problem. Whenever an operator is added or modified, it changes `ATen/Functions.h` and `ATen/NativeFunctions.h` which in turn requires essentially every single file to be rebuilt. Per-operator headers allow files to only include the specific operators they use and so minimizes unnecessary rebuilds during incremental builds and improves cache hits in CI builds.

See this note for more details:
https://github.com/pytorch/pytorch/blob/3a03af2f505494582f1359254431f0c47fd88705/aten/src/ATen/templates/Functions.h#L20

Differential Revision: [D33949900](https://our.internmc.facebook.com/intern/diff/D33949900)

[ghstack-poisoned]

malfet commented:
@malfet has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D33949900).
